src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -622,71 +626,138 @@ private AIState waitForShovel()
+        if (isToolInHut(toolType, minimalLevel))
         {
             return false;
         }
         if (!getOwnBuilding().hasOnGoingDelivery())
         {
-            chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_TOOLREQUEST, tool, InventoryUtils.swapToolGrade(hutLevel));
+            chatRequestTool(toolType, minimalLevel, maxToolLevel);
+
         }
         return true;
     }
 
+    private void chatRequestTool(@NotNull final ToolType toolType, final int minimalLevel, final int maximumLevel)
+    {
+        switch(toolType)
+        {
+            case PICKAXE:
+                if (minimalLevel > maximumLevel)
+                {
+                    chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_PICKAXEREQUESTBETTERHUT,
+                      ItemStackUtils.swapToolGrade(maximumLevel));
+                }
+                else if (minimalLevel == maximumLevel)
+                {
+                    chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_PICKAXEREQUEST,
+                      ItemStackUtils.swapToolGrade(maximumLevel));
+                }
+                else
+                {
+                    chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_PICKAXEREQUEST,
+                      ItemStackUtils.swapToolGrade(minimalLevel),
+                      ItemStackUtils.swapToolGrade(maximumLevel));
+                }
+                break;
+            case BOW:
+            case FISHINGROD:
+                if (maximumLevel<1)
+                {
+                    chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_SIMPLETOOLREQUEST, toolType.getName());
+                }
+                else
+                {
+                    chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_ENCHTOOLREQUEST, toolType.getName(), maximumLevel-1);
+                }
+                break;
+            default:
+                chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_TOOLREQUEST, toolType.getName(), ItemStackUtils.swapToolGrade(maximumLevel));
+                break;
+        }
+    }
+
     /**
      * Check all chests in the worker hut for a required tool.
      *
-     * @param tool the type of tool requested (amount is ignored)
+     * @param toolType the type of tool requested (amount is ignored)
+     * @param minimalLevel the minimal level the tool should have.
      * @return true if a stack of that type was found
      */
-    public boolean isToolInHut(final String tool)
+    public boolean isToolInHut(final ToolType toolType, final int minimalLevel)
     {
         @Nullable final AbstractBuildingWorker building = getOwnBuilding();
 
-        boolean hasItem;
         if (building != null)
         {
-            hasItem = isToolInTileEntity(building.getTileEntity(), tool);
-
-            if (hasItem)
+            //TODO check last 2 parameters
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)